pkg/lifecycle-poc/funnel/connector_metrics.go

```diff
@@ -0,0 +1,73 @@
+// Copyright © 2025 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package funnel
+
+import (
+	"strings"
+	"time"
+
+	"github.com/conduitio/conduit-commons/opencdc"
+	"github.com/conduitio/conduit/pkg/connector"
+	"github.com/conduitio/conduit/pkg/foundation/metrics"
+	"github.com/conduitio/conduit/pkg/foundation/metrics/measure"
+)
+
+type ConnectorMetrics interface {
```

Comment:
I didn't feel to creative with the interface and struct names.:) I'll think of something prettier.

Reply:
@hariso I actually think it's not that bad. Especially considering that when we use it looks like `t.metrics.Observe(records[len(acks):len(acks)+len(acksResp)], start)` via `metrics ConnectorMetrics`.